external/storm-kafka-client/src/main/java/org/apache/storm/kafka/KafkaUtils.java

```diff
@@ -0,0 +1,114 @@
+
+                    for (Map.Entry<TopicPartition, OffsetManager> entry : offsetManagers.entrySet()) {
+                        TopicPartition topicPartition = entry.getKey();
+                        OffsetManager offsetManager = entry.getValue();
+
+                        long latestTimeOffset = beginningOffsets.get(topicPartition);
+                        long earliestTimeOffset = endOffsets.get(topicPartition);
+
+                        long latestEmittedOffset = offsetManager.getLatestEmittedOffset();
+                        long latestCompletedOffset = offsetManager.getCommittedOffset();
+                        long spoutLag = latestTimeOffset - latestCompletedOffset;
+
+                        String metricPath = topicPartition.topic()  + "/partition_" + topicPartition.partition();
+                        ret.put(metricPath + "/" + "spoutLag", spoutLag);
+                        ret.put(metricPath + "/" + "earliestTimeOffset", earliestTimeOffset);
+                        ret.put(metricPath + "/" + "latestTimeOffset", latestTimeOffset);
+                        ret.put(metricPath + "/" + "latestEmittedOffset", latestEmittedOffset);
+                        ret.put(metricPath + "/" + "latestCompletedOffset", latestCompletedOffset);
+
+                        if (!topicMetricsMap.containsKey(topicPartition.topic())) {
+                            topicMetricsMap.put(topicPartition.topic(),new TopicMetrics());
+                        }
+
+                        TopicMetrics topicMetrics = topicMetricsMap.get(topicPartition.topic());
+                        topicMetrics.totalSpoutLag += spoutLag;
+                        topicMetrics.totalEarliestTimeOffset += earliestTimeOffset;
+                        topicMetrics.totalLatestTimeOffset += latestTimeOffset;
+                        topicMetrics.totalLatestEmittedOffset += latestEmittedOffset;
+                        topicMetrics.totalLatestCompletedOffset += latestCompletedOffset;
+                    }
+
+                    for(Map.Entry<String, TopicMetrics> e : topicMetricsMap.entrySet()) {
+                        String topic = e.getKey();
+                        TopicMetrics topicMetrics = e.getValue();
+                        ret.put(topic + "/" + "totalSpoutLag", topicMetrics.totalSpoutLag);
+                        ret.put(topic + "/" + "totalEarliestTimeOffset", topicMetrics.totalEarliestTimeOffset);
+                        ret.put(topic + "/" + "totalLatestTimeOffset", topicMetrics.totalLatestTimeOffset);
+                        ret.put(topic + "/" + "totalLatestEmittedOffset", topicMetrics.totalLatestEmittedOffset);
+                        ret.put(topic + "/" + "totalLatestCompletedOffset", topicMetrics.totalLatestCompletedOffset);
+                    }
+
+                    return ret;
+                } else {
+                    LOG.info("Metrics Tick: Not enough data to calculate spout lag.");
+                }
+            } catch (Throwable t) {
```

Comment:
Why are we catching all exceptions here?

Reply:
Removed Throwable from catch clause.